include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
@gdr-at-ms That particular test is already inside a test for `_MSC_VER` (or at least I thought it was). So the intent was to test for both the presence of MSVC (via `_MSC_VER`) and the presence of `std::byte` (via the MSVC STL feature flag).